app/utils/Executor.py

```diff
@@ -74,33 +74,37 @@ def killer(process):
 
     def workflow(self, patch: Patch):
         assert self.__current_patch is None, 'no auto-concurrency!'
-        self.__current_patch = patch
-
-        # step 1: write patch to temp file
-        patchfile = tempfile.NamedTemporaryFile(delete=False, mode='w')
-        patchfile.write(patch.patch)
-        patchfile.close()
-
-        # step 2: apply patch
-        self.__execute_command_timeout('patch -p1 --input={patchfile}'.format(patchfile=patchfile.name), cwd='/')
-
-        # step 3: command pipeline
-        success = self.__apply_command(patch, 'build_command') and \
-                  self.__apply_command(patch, 'quickcheck_command') and \
-                  self.__apply_command(patch, 'test_command')
-
-        if success:
-            patch.state = 'survived'
-            db.session.commit()
-
-        # step 4: revert patch
-        self.__execute_command_timeout('patch -p1 --reverse --input={patchfile}'.format(patchfile=patchfile.name),
+        
+        file = File.query.get(patch.file_id)
+        
+        if file is not None:
+            self.__current_patch = patch
+
+            # step 1: write patch to temp file
+            patchfile = tempfile.NamedTemporaryFile(delete=False, mode='w')
+            patchfile.write(patch.patch)
+            patchfile.close()
+
+            # step 2: apply patch
+            self.__execute_command_timeout('patch -p1 --input={patchfile} {inputfile}'.format(patchfile=patchfile.name, inputfile=file.filename), cwd='/')
+
+            # step 3: command pipeline
+            success = self.__apply_command(patch, 'build_command') and \
+                      self.__apply_command(patch, 'quickcheck_command') and \
+                      self.__apply_command(patch, 'test_command')
+
+            if success:
+                 patch.state = 'survived'
+                 db.session.commit()
+
+            # step 4: revert patch
+            self.__execute_command_timeout('patch -p1 --reverse --input={patchfile} {inputfile}'.format(patchfile=patchfile.name, inputfile=file.filename),
                                        cwd='/')
```

Comment:
What is the rationale for this changed block?

Reply:
This is needed to work properly on Ubuntu 16.04. If launch ```patch -p1 --reverse --input={patchfile}``` then patch will fail and request file name to patch. And with ```patch -p1 --reverse --input={patchfile} {inputfile}``` all works as expected.